### ISSUE TITLE
loading external libraries from git is broken

### ISSUE THREAD

**Issue by spencerking** [carp-lang/Carp#595]
I have been trying to load the file library from carpentry using the following code:

```
(load "git@github.com:carpentry-org/file@0.0.4")

(File.open "test.csv")

(IO.println &(read-all &f))
```

This gives me the following error:
```
I can't find a file named: '/Users/spencerking/.carp/libs//github.com_COLON_carpentry-org/file/0.0.4/main.carp'

If you tried loading an external package, try appending a version string (like `@master`)
```

I initially noticed the "//" after libs and corrected this by editing app/Main.hs to remove the trailing slash in the projectLibDir variable. I recompiled carp and tried again, receiving the same error minus the double slash. 

It was at this point I realized the file library had not actually been downloaded into .carp.

Is this an actual bug, or am I just missing a step? If this is a bug, I would be happy to try fixing it if someone could point me in the right direction.

**Comment by hellerve** [comment]
Thanks for reporting this!

As a quick fix for this you can get the library via https instead (using `https://github.com/carpentry-org/file@0.0.4`).

Carp needs to propagate this Git error, though. It’s curious that it didn’t. I’ll look into it!

**Comment by hellerve** [comment]
I think I know what this is: I get the correct error the first time, but we still create the directory we want to clone in. By the second time we try to clone, this directory path exists, we assume it was already downloaded last time, and we try to load from it but are unable to because it failed last time.

This needs a bit of love, but I think I’ll be able to provide a PR that resolves this!

**Comment by spencerking** [comment]
Yes, you are correct. I deleted the directories and am now seeing that git error on the initial attempt. I must have missed it the first time.
